app/iot/service.py

```diff
@@ -26,24 +25,25 @@ class IOTService:
     def __init__(self) -> None:
         self.devices: dict[str, Device] = {}
 
-    def register_device(self, device: Device) -> str:
+    async def register_device(self, device: Device) -> str:
         device.connect()
```

Comment:
Here must use await
If you make async function, then must be some await inside)